### PR TITLE
Implement align-model attribute

### LINKED ISSUE #417: Configuration to control if model is centered at root origin or bounding box origin
Currently, we stage a model by determining its bounding box, and centering it in the viewport based on its bounding box's origin. This behavior is a reasonable default for a lot of scenarios.

Sometimes it is desirable to center a model at the origin of the root node of its scene graph. We should add a configuration that allows the user to specify this centering strategy.

This is semi-related to #271, where @jsantell mentions that this configuration might be desirable in the AR placement context.

### LINKED ISSUE #440: Materials not being cloned correctly
In #421 we introduced an updated cloning routine for glTFs, and merged it with our previous behavior of cloning all materials in the scene.

Unfortunately, I introduced a regression that causes materials not to be cloned in the majority of circumstances, resulting in a lot of bleed-over related to lighting configuration across different copies of the same model.

### PR DESCRIPTION
**Pull request by cdata**
# `align-model` attribute

<div align="center">
  
</div>

---

This change proposes a new configurable attribute `align-model` that can change how the model is aligned relative to the scene. Important highlights include:

 - `align-model` accepts values `center` and `origin`
   - `center` refers to the center of the bounding volume of the model
   - `origin` refers to the origin of the model relative to world space
   - Accepts two- or three-value formats in the spirit of CSS properties like [`transform-origin`](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-origin)
   - For two-value form, the first value is also used to fill in the third value
 - `align-model` and `auto-rotate` have been conceptually grouped together as the "staging" feature
 - Fixed a bug related to bleeding lighting configuration (see #440)
 - A handful of additional memory leak sources were squashed in the test suite

**🛂Note for reviewers:** I have spent _a lot_ of time noodling on the names. I do _not_ love any of them. Please don't hesitate to bike shed!

### Different values of `align-model`

`center` (default) | `origin` | `origin center` | `center origin center`
------------------ | -------- | --------------- | --------------------
image | image | image | image

Fixes #417 
Fixes #440 
Includes further mitigations related to #433

### CODE REVIEW

File: src/three-components/Renderer.js [hunk 1]
```diff
@@ -191,6 +174,11 @@ export default class Renderer extends EventDispatcher {
         this.setRendererSize(maxWidth, maxHeight, false);
       }
 
+      const {exposure} = scene;
+      const exposureIsNumber =
+          typeof exposure === 'number' && !self.isNaN(exposure);
+      this.renderer.toneMappingExposure = exposureIsNumber ? exposure : 1.0;
```
Comment (jsantell) [inline src/three-components/Renderer.js]: Good catch here too, exposure is scene specific

File: src/model-viewer.js [hunk 2]
```diff
@@ -15,20 +15,21 @@
...
 
 const ModelViewerElement = [
   AnimationMixin,
   LoadingMixin,
   ARMixin,
-  AutoRotateMixin,
+  StagingMixin,
   ControlsMixin,
   EnvironmentMixin,
+  StagingMixin,
```
[the remark this answers is not in the document]
Comment (cdata) [reply src/model-viewer.js]: Good catch, will fix!

File: src/three-components/ModelScene.js [hunk 3]
```diff
@@ -259,8 +285,12 @@ export default class ModelScene extends Scene {
     // @see ROOM_PADDING_SCALE
     scale /= ROOM_PADDING_SCALE;
 
+    modelCenter.multiply(this[$modelAlignmentMask]);
+    this.unscaledModelOffset.copy(modelCenter).multiplyScalar(-1);
     modelCenter.multiplyScalar(scale);
+
     this.model.scale.multiplyScalar(scale);
+    this.model.position.copy(roomCenter);
```
[the remark this answers is not in the document]
Comment (cdata) [reply src/three-components/ModelScene.js]: Will fix

**Comment by smalls** [comment]
:bike: :derelict_house: ... Sorry I didn't get this into #417 while we were still discussing there.

In some ways, at least as I look at the illustration above, it seems as though what the transformation is doing is changing how the scene is composed around the model. I wonder if an attribute of `scene-alignment` or `scene-origin`, with values `model-center` or `model-origin` might communicate that?

**Comment by cdata** [comment]
@smalls thanks for the feedback! At this point the most useful thing is probably to file an issue so that we revisit this before the next major release

**Pull request merged**